### PR TITLE
Process network errors

### PR DESCRIPTION
**Pull request by sadym-chromium**
Initial implementation of the `Network.fetchError`. Not fully compliant with the spec, but pushing anyways to avoid hanging event queue eg. like in #605.

### CODE REVIEW

File: tests/test_network.py [hunk 1]
```diff
@@ -241,3 +241,42 @@ async def test_network_network_response_completed_event_emitted(
             }
         }
     }
+
+
+@pytest.mark.asyncio
+async def test_network_bad_ssl(websocket, context_id):
+    BAD_SSL_URL = "https://expired.badssl.com/"
```
Comment (thiagowfx) [inline tests/test_network.py:248]: We have a bunch of TODOs to make the tests work offline. Does it make sense to add one here as well?

File: tests/test_helpers.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from __future__ import annotations
```
Comment (thiagowfx) [inline tests/test_helpers.py:15]: Why is this needed?

**Review by thiagowfx — APPROVED** [review]

File: tests/test_helpers.py (hunk 2 above)
Comment (sadym-chromium) [reply tests/test_helpers.py:15]: In my setup it is needed to use types:
```
test_helpers.py:138: in <module>
    def AnyExtending(expected: list | dict):
E   TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

File: tests/test_network.py (hunk 1 above)
Comment (sadym-chromium) [reply tests/test_network.py:248]: added

**Review by thiagowfx — APPROVED** [review]

**Comment by Lightning00Blade** [comment]
FYI this produces an unhandled error atm:
```
Uncaught Error: the object {
  "canceled": false
  "errorText": "net::ERR_CERT_AUTHORITY_INVALID"
  "requestId": "6734277685B2C23C5D68E738D948F08A"
  "timestamp": 473342.498087
  "type": "Document"
} was thrown, throw an Error :)
```
Please revisit when you do further implementations here.
